AsyncDisplayKit/Layout/ASLayout.h

```diff
@@ -21,20 +21,39 @@ extern BOOL CGPointIsNull(CGPoint point);
 /** Represents a computed immutable layout tree. */
 @interface ASLayout : NSObject
 
+/**
+ * The underlying object described by this layout
+ */
 @property (nonatomic, weak, readonly) id<ASLayoutable> layoutableObject;
+
+/**
+ * Size of the current layout
+ */
 @property (nonatomic, readonly) CGSize size;
+
 /**
  * Position in parent. Default to CGPointNull.
  * 
  * @discussion When being used as a sublayout, this property must not equal CGPointNull.
  */
 @property (nonatomic, readwrite) CGPoint position;
-/** 
+
+/**
  * Array of ASLayouts. Each must have a valid non-null position.
  */
 @property (nonatomic, readonly) NSArray<ASLayout *> *sublayouts;
 
 /**
+ * A list of sublayouts that were not already flattened.
+ */
+@property (nonatomic, readonly) NSArray<ASLayout *> *immediateSublayouts;
```

Comment:
Does this need to be retained for the rest of the node lifecycle to support transitions?  If so, main concern is that it not lose its final reference when deallocating the node on the main thread, since we must dealloc all nodes there in order to ensure developers' code is safe (as the node may have a gesture recognizer or UIView type instance variable)